core/math/a_star_grid_2d.cpp

```diff
@@ -123,13 +123,18 @@ AStarGrid2D::CellShape AStarGrid2D::get_cell_shape() const {
 
 void AStarGrid2D::update() {
 	points.clear();
+	solid_mask.clear();
 
 	const int32_t end_x = region.get_end().x;
 	const int32_t end_y = region.get_end().y;
 	const Vector2 half_cell_size = cell_size / 2;
+	for (int32_t x = region.position.x; x < end_x + 2; x++) {
+		solid_mask.push_back(true);
```

Comment:
This is to create an artificial bound around to eliminate the need for checking the region begin/end? Smart decision.

Reply:
That is exactly it! thank you!